daemon/pod.go

```diff
@@ -42,11 +43,201 @@ type Pod struct {
 	sync.RWMutex
 }
 
-func NewPod(rawSpec []byte, id string, data interface{}) (*Pod, error) {
+// TODO: remove convertToRunvPodSpec after pod.UserPod is deleted from runv
+func convertToRunvPodSpec(podSpec *apitypes.UserPod) (*pod.UserPod, error) {
```

Comment:
> If user uses grpc api and pod spec to create pod, the extra conversion from json to protobuf is needed.

No, there is no json to protobuf conversion in gRPC api. Users could pass apitypes.UserPod struct directly in create API.